### DISCUSSION TITLE
Conditional join() based on node name

### DISCUSSION THREAD

**Discussion by jbrialon** [donmccurdy/glTF-Transform#1571] [Q&A] [answered]
Hello everyone,

I would like to use the join() transform to reduce the draw calls but i'm animating some parts of the model that are joined, I would like to know if there is a way to conditionally join() the meshes based on the node name so I can excluse the one I want to animate

thanks in advance

**Comment by donmccurdy** [comment, marked as answer]
Hi! At present I think the way to do this would be to remove the names from any meshes you consider 'candidates' for merging (keeping names on those you want to animate) and then run join with the 'keepNamed' option enabled. Full options at:

https://gltf-transform.dev/modules/functions/interfaces/JoinOptions

Alternatively you can join exactly the primitives you want, with `joinPrimitives`:

https://gltf-transform.dev/modules/functions/functions/joinPrimitives

**Reply by jbrialon** [reply]
ooh thanks a lot, I did not think of that, this is working thank you 👍
